packages/iris-grid/src/sidebar/conditional-formatting/ConditionEditor.tsx

```diff
@@ -57,12 +65,185 @@ const dateConditions = [
   DateCondition.IS_BEFORE_OR_EQUAL,
   DateCondition.IS_AFTER,
   DateCondition.IS_AFTER_OR_EQUAL,
+  DateCondition.IS_NULL,
+  DateCondition.IS_NOT_NULL,
 ].map(option => (
   <option key={option} value={option}>
     {getLabelForDateCondition(option)}
   </option>
 ));
 
+const booleanConditions = [
+  BooleanCondition.IS_TRUE,
+  BooleanCondition.IS_FALSE,
+  BooleanCondition.IS_NULL,
+  BooleanCondition.IS_NOT_NULL,
+].map(option => (
+  <option key={option} value={option}>
+    {getLabelForBooleanCondition(option)}
+  </option>
+));
+
+const charConditions = [
+  CharCondition.IS_EQUAL,
+  CharCondition.IS_NOT_EQUAL,
+  CharCondition.IS_NULL,
+  CharCondition.IS_NOT_NULL,
+].map(option => (
+  <option key={option} value={option}>
+    {getLabelForCharCondition(option)}
+  </option>
+));
+
+function isNumberConditionValid(
+  condition: NumberCondition,
+  value?: string | number,
+  startValue?: string | number,
+  endValue?: string | number
+) {
+  if (
+    condition === NumberCondition.IS_NULL ||
+    condition === NumberCondition.IS_NOT_NULL
+  ) {
+    return true;
+  }
+  if (
+    condition === NumberCondition.IS_BETWEEN &&
+    !Number.isNaN(Number(startValue)) &&
+    !Number.isNaN(Number(endValue))
+  ) {
+    return true;
+  }
+  if (
+    condition !== NumberCondition.IS_BETWEEN &&
+    !Number.isNaN(Number(value))
+  ) {
+    return true;
+  }
+  return false;
+}
+
+function getNumberInputs(
+  selectedCondition: NumberCondition,
+  handleValueChange: (e: unknown) => void,
+  handleStartValueChange: (e: unknown) => void,
+  handleEndValueChange: (e: unknown) => void,
+  conditionValue?: string | number,
+  startValue?: number,
+  endValue?: number
+) {
+  switch (selectedCondition) {
+    case NumberCondition.IS_EQUAL:
+    case NumberCondition.IS_NOT_EQUAL:
+    case NumberCondition.GREATER_THAN:
+    case NumberCondition.GREATER_THAN_OR_EQUAL:
+    case NumberCondition.LESS_THAN:
+    case NumberCondition.LESS_THAN_OR_EQUAL:
+      return (
+        <input
+          type="text"
```

Comment:
This might be a dumb question... should we use `type="number"` here?

Reply:
Good point, updated.